crates/bevy_render/src/fly_camera.rs

```diff
@@ -0,0 +1,184 @@
+use crate::{
+    camera::{Camera, PerspectiveProjection, VisibleEntities},
+    render_graph::base,
+};
+use bevy_app::{AppBuilder, EventReader, Events, Plugin};
+use bevy_core::Time;
+use bevy_ecs::*;
+use bevy_ecs::{Bundle, Query, Res, ResMut};
+use bevy_input::{keyboard::KeyCode, mouse::MouseMotion, Input};
+use bevy_math::{Quat, Vec2, Vec3};
+use bevy_transform::components::{Rotation, Scale, Transform, Translation};
+
+/**
+Used in [CameraFlyingComponents](struct.CameraFlyingComponents.html)
+**/
+pub struct CameraFlyingOptions {
+    /// The camera's flying speed
+    pub speed: f32,
+    /// The camera's mouse sensitivity
+    pub sensitivity: f32,
+    /// The vertical pitch of the camera, constrained between `89.9f32` and `-89.9f32`. This value is kept up-to-date by the [CameraFlyingPlugin](struct.CameraFlyingPlugin.html) plugin, and can be mutated to adjust the camera's angle
+    pub pitch: f32,
+    /// The horizontal yaw of the camera. This value is kept up-to-date by the [CameraFlyingPlugin](struct.CameraFlyingPlugin.html) plugin, and can be mutated to adjust the camera's angle
+    pub yaw: f32,
+}
+impl Default for CameraFlyingOptions {
+    fn default() -> Self {
+        Self {
+            speed: 10.0,
+            sensitivity: 10.0,
+            pitch: 0.0,
+            yaw: 0.0,
+        }
+    }
+}
+
+/**
+A basic flying camera for 3D scenes.

```

Comment:
Thanks for adding nice documentation here!

Reply:
👍